File: examples/primitives/python/src/templates/template_functions_pygen.py
from __future__ import annotations
from typing import *
import examples.templates.template_methods as pybind_template_methods_pygen
from bind_utils import *
from metaclass import *
from simple import *



class TemplateFunctions(metaclass=OriginalMethodsMetaclass):
    """
    comments
    """
    
    @bind
    def __init__(self):
        pass
    
    @bind
    def max(self, arg0: int, arg1: int) -> int:
        """
        comments
        """
        pass

    @bind
    def max(self, arg0: str, arg1: str) -> str:
        """
        comments
        """
        pass

    @bind
    def make_pair(self, arg0: Project, arg1: Project) -> tuple:
        """
        comments
        """
        pass

    @bind
    def make_pair(self, arg0: PyTask, arg1: Project) -> tuple:
        """
        comments
        """
        pass

File: examples/primitives/python/wrappers/templates/template_functions_pygen.cpp
#include <pybind11/pybind11.h>
#include <pybind11/stl.h>
#include <pybind11/functional.h>

#include "cxx/templates/template_functions.hpp"
#include "cxx/simple/project.hpp"
#include "cxx/simple/task.hpp"

namespace py = pybind11;

void bindTemplateMethods(py::module& m) {
    using namespace iegen::example;
    py::class_<iegen::example::TemplateMethods> templatemethods(m, "TemplateMethods");

    templatemethods.def(py::init<>());
    templatemethods.def("max", &iegen::example::TemplateMethods::max<int>, py::arg("arg0"), py::arg("arg1"));
    templatemethods.def("max", &iegen::example::TemplateMethods::max<std::string>, py::arg("arg0"), py::arg("arg1"));
    templatemethods.def("make_pair", &iegen::example::TemplateMethods::makePair<iegen::example::Project, iegen::example::Project>, py::arg("arg0"), py::arg("arg1"));
    templatemethods.def("make_pair", &iegen::example::TemplateMethods::makePair<iegen::example::Task, iegen::example::Project>, py::arg("arg0"), py::arg("arg1"));
}